Configurations/windows-makefile.tmpl

```diff
@@ -354,7 +356,7 @@ uninstall: uninstall_docs uninstall_sw
 
 libclean:
 	"$(PERL)" -e "map { m/(.*)\.dll$$/; unlink glob """{.,apps,test,fuzz}/$$1.*"""; } @ARGV" $(SHLIBS)
-	-del /Q /F $(LIBS) libcrypto.* libssl.* ossl_static.pdb
+	-del /Q /F $(LIBS) libcrypto*.* libssl*.* ossl_static*.pdb
```

Comment:
We *could* make better use of `$(SHLIBS)` and `$(SHLIBPDBS)`, no?

Reply:
Should `shlib_variant` be renamed to `lib_variant` as at least on windows it meant to change the name of the static version as well? Is dynamic loading used within the OpenSSL project itself? If the engine dll's do not have different names then how can they co-exist in the same directory? By looking at the [FindOpenSSL.cmake](https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake#L152) at least for debug and release the same paths are expected and lib/dll names differ in the suffix: "d" for the debug variant and "" for release. I do not like the CMake module at all. I would really prefer 4 different sub-folders with a fix naming scheme where each contains an unmodified OpenSSL install. But a change there will probably break far more projects than adding an additional OpenSSL build target. For me it is not clear what the ideal directory structure would look like and what the names of the different variants would be.
Maybe if the OpenSSL Project defines official locations and a structure on windows, it is enough for people that offer precompiled libraries and the CMake project to follow.  

I try to find a bit more time to learn more about linking and library resolving especially on windows. Thank you for pushing me in the right direction.